src/frontend/apps/impress/src/features/docs/docs-grid/components/DocsGridActions.tsx

```diff
@@ -10,29 +15,32 @@ interface DocsGridActionsProps {
 
 export const DocsGridActions = ({ doc }: DocsGridActionsProps) => {
   const { t } = useTranslation();
-  const [isModalRemoveOpen, setIsModalRemoveOpen] = useState(false);
+  const dropdownMenu = useDropdownMenu();
+  const deleteModal = useModal();
 
-  if (!doc.abilities.destroy) {
-    return null;
-  }
+  const options: DropdownMenuOption[] = [
+    {
+      label: t('Remove'),
+      icon: 'delete',
+      callback: () => deleteModal.open(),
+      disabled: !doc.abilities.destroy,
+      testId: `docs-grid-actions-remove-${doc.id}`,
+    },
+  ];
 
   return (
     <>
-      <Button
-        data-testid={`docs-grid-delete-button-${doc.id}`}
-        onClick={(event) => {
-          event.preventDefault();
-          event.stopPropagation();
-          setIsModalRemoveOpen(true);
-        }}
-        color="tertiary-text"
-        icon={<span className="material-icons">delete</span>}
-        size="small"
-        style={{ padding: '0rem' }}
-        aria-label={t('Delete the document')}
-      />
-      {isModalRemoveOpen && (
-        <ModalRemoveDoc onClose={() => setIsModalRemoveOpen(false)} doc={doc} />
+      <DropdownMenu {...dropdownMenu} options={options}>
+        <Icon
+          data-testid={`docs-grid-actions-button-${doc.id}`}
+          iconName="more_horiz"
+          $theme="primary"
+          $variation="600"
+        />
+      </DropdownMenu>
```

Comment:
Yes but in a following PR, we will add the options for download. So there will definitely be the button. However, we can ask Robin to indicate on the line whether we have editing rights.

Reply:
It's on Robin's todo to know at a glance if we can do admin actions

Reply:
Let's consider this from a user's perspective:

1. I browse through my list of documents
2. I notice some documents are obsolete (unused or completed projects)
3. I want to remove these documents to maintain a cleaner, more focused list

Feature-wise, here's a straightforward solution:

- If the user is owner/admin, they can delete the document
- Otherwise, the user can leave the document

In practice, the delete button simply transforms into a leave button.

What do you think?

Reply:
That is quite nice, because yes, actually we don't have a way for the moment to remove (leave) a document when we are not owner/admin.
We need to add a DELETE endpoint on the backend side, to remove the LinkTrace entry:
https://github.com/numerique-gouv/impress/blob/f596c99a2165e4ddbc48a7fa2fc95509ebabb9c8/src/backend/core/api/viewsets.py#L396-L400